resources/benchmark-configurator.mjs

```diff
@@ -0,0 +1,149 @@
+                    return;
+                }
+
+                const response = await fetch(params.config);
+
+                if (!response.ok)
+                    throw new Error(`Could not fetch config: ${response.status}`);
+
+                const config = await response.json();
+
+                if (!config || !Array.isArray(config.suites))
+                    throw new Error("Could not find a valid config structure!");
+
+                config.suites.flatMap((suite) => suite.tags || []).forEach((tag) => this.#tags.add(tag));
+                config.suites.forEach((suite) => {
+                    if (suite && suite.url && this._isValidUrl(suite.url))
+                        this.#suites.push(suite);
+                    else
+                        throw new Error("Invalid suite data");
+                });
+            } catch (error) {
+                console.warn(`Error loading custom configuration: ${error.message}. Loading default suites.`);
+                this._loadDefaultSuites();
+            }
+        } else {
+            this._loadDefaultSuites();
+        }
+
+        this._freezeTags();
+        this._freezeSuites();
+    }
+
+    _loadDefaultSuites() {
+        defaultSuites.flatMap((suite) => suite.tags).forEach((tag) => this.#tags.add(tag));
+        defaultSuites.forEach((suite) => this.#suites.push(suite));
+    }
+
+    enableSuites(names, tags) {
+        if (names?.length) {
+            const lowerCaseNames = names.map((each) => each.toLowerCase());
+            this.#suites.forEach((suite) => {
+                suite.enabled = lowerCaseNames.includes(suite.name.toLowerCase());
+            });
+        } else if (tags?.length) {
+            tags.forEach((tag) => {
+                if (!this.#tags.has(tag))
+                    console.error(`Unknown Suites tag: "${tag}"`);
+            });
+            const tagsSet = new Set(tags);
+            this.#suites.forEach((suite) => {
+                suite.enabled = suite.tags.some((tag) => tagsSet.has(tag));
+            });
+        } else {
+            console.warn("Neither names nor tags provided. Enabling all default suites.");
+            this.#suites.forEach((suite) => {
+                suite.enabled = suite.tags.includes("default");
+            });
+        }
+        if (this.#suites.some((suite) => suite.enabled))
+            return;
+        let message, debugInfo;
+        if (names?.length) {
+            message = `Suites "${names}" does not match any Suite. No tests to run.`;
+            debugInfo = {
+                providedNames: names,
+                validNames: this.#suites.map((each) => each.name),
+            };
+        } else if (tags?.length) {
+            message = `Tags "${tags}" does not match any Suite. No tests to run.`;
+            debugInfo = {
+                providedTags: tags,
+                validTags: Array.from(this.#tags),
+            };
+        }
+        alert(message);
```

Comment:
I think we should extra a helper function like _reportError which takes message & debugInfo instead of declaring local variables then if-else branching it like this. So that we can do instead:
```js
if (names?.length)
    this._reportError(`Suites "${names}" does not match any Suite. No tests to run.`, { providedNames: names, validNames: this.#suites.map((each) => each.name) });
else
    this._reportError(`Tags "${tags}" does not match any Suite. No tests to run.`, { providedTags: tags, validTags: Array.from(this.#tags) });
```
It could be a local lambda too.